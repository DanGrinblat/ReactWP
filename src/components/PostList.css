.post-list {
	display: grid;
	position: absolute;
	top: 50%;
	grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
	grid-gap: 0 10px;
	grid-auto-rows: 10px;
	vertical-align: middle;
}

/* Make room for pagination module and footer */
.ui.grid {
	margin-top: 0 !important;
	margin-bottom: -58px !important;
	min-height: calc(100vh - 100px);
}
