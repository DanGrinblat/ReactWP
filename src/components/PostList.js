import "./PostList";
import React from "react";
import Post from "./Post";

class PostList extends React.Component {
	renderList() {
		return this.props.posts.map(post => {
			for (var i in this.props.media) {
				if (post.featured_media === this.props.media[i].id) {
					post.imageLink = this.props.media[i].source_url;
				}
			}

			return (
				<div key={post.id} className="equal width column">
					<center>
						<Post post={post} />
					</center>
				</div>
			);
		});
	}

	render() {
		return (
			<div className="ui container three column doubling stackable middle aligned centered grid post-list">
				{this.renderList()}
			</div>
		);
	}
}

export default PostList;
