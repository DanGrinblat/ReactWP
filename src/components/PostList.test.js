import React from "react";
import { shallow } from "enzyme";
import PostList from "./PostList";

describe("PostList", () => {
	const posts = [
		{
			id: 1,
			featured_media: null,
			imageLink: null,
			title: "title",
			excerpt: { rendered: "excerpt" }
		}
	];

	const postList = shallow(<PostList posts={posts} />);

	it("should render correctly", () => {
		expect(postList).toMatchSnapshot();
	});
});
