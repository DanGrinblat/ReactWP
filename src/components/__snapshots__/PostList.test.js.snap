// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList should render correctly 1`] = `
<div
  className="ui container three column doubling stackable middle aligned centered grid post-list"
>
  <div
    className="equal width column"
    key="1"
  >
    <center>
      <Link
        to="/blog/undefined"
      >
        <PostSummary
          post={
            Object {
              "excerpt": Object {
                "rendered": "excerpt",
              },
              "featured_media": null,
              "id": 1,
              "imageLink": null,
              "title": "title",
            }
          }
        />
      </Link>
    </center>
  </div>
</div>
`;
